woocommerce/woocommerce
# [Accessibility] Add required a11y text and decoration to required fields

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #48948 .
- Update forms that the `wc-template-functions` and Address changes to ensure that the asterisk is seen but not read by a screen reader (`aria-required="true"` will announce that the field is required) 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open `/my-account/edit-account/` and using a screen reader, tab to each field. Make sure that the screen reader reads each fields label, required (if required), along with any descriptive text below the fields.
2. Edit each of the addresses `/my-account/edit-address/billing/` and `/my-account/edit-address/shipping/`, and follow the same process as you did in step 1, making sure that the screen reader correctly reads each field as you tab through the form.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
